src/components/elements/metadataKeyValue/MetadataKeyValueList.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Box from '@mui/material/Box';
+import {MetadataKeyValuePair, MetadataKeyValuePairProps} from './MetadataKeyValuePair';
+import StyledTextField from './StyledTextField';
+import IconButton from '@mui/material/IconButton';
+import AddIcon from '@mui/icons-material/Add';
+
+export interface MetadataKeyValueListProps {
+  maxMetadataFieldsSectionHeight?: string;
+  metadataList: {
+    key: string;
+    value: string;
+    valueType: string;
+    multiple: boolean;
+    isAutoGenerated?: boolean;
+  }[];
+  isEditable: boolean;
+  deletionEnabled: boolean;
+  onDeleteHandler?: (keyName:string)=>void;
+  onChangeHandler: (
+    metadataList: {
+      key?: string;
+      value?: string;
+      valueType?: string;
+      multiple: boolean;
+      isAutoGenerated?: boolean;
+    }[]
+  ) => void;
+}
+
+export function MetadataKeyValueList({
+  maxMetadataFieldsSectionHeight,
+  metadataList,
+  isEditable,
+  deletionEnabled,
+  onDeleteHandler,
+  onChangeHandler
+}: MetadataKeyValueListProps) {
+  //Todo:
+  // - Not sure what to do with the multiple field. (If I need to use it as part of the validations in the future, and also what value should I set for newly created fields).
+  // - Validations that are missing:
+  //  * Check whether new field created was left empty
+  //  * When edit/ creating new keys, make sure the value match the value type.
+  //  * When creating new keys, make sure that the new key name doesn't already exist.
+  // - Not sure yet how the external save progress should work. For now, whenever I'm saving changes locally, I'm calling the external 'onChangeHandler' func.
+  //    Whoever use this component will have external save button.
+  // - The deletion of existing metadata field is immediate, and do not depend on the external save button
```

Comment:
This should trigger an external function, for which deletion from the component should succeed only if the external function returned a success value.
For example, 
```
if (onDeleteMetadata(field) { 
  <actually delete> 
} else { 
  alert("failed to delete") 
}
```
Or catch exception or whatever makes sense, we need to decide on a convention for how we do backend validations like that.